### PR TITLE
[feature][connector] ElasticSearch Sink: option to output canonical key fields (JSON and Avro)

### PR DESCRIPTION
**Pull request by nicoloboschi**
### Motivation
In case of the inbound message is structured (Avro or JSON) the fields order may change overtime and there's no fields order guarantee from the Pulsar function framework.
The generated _id field of the document is supposed to be the same regardless the input message key fields order.    

### Modifications
* New option `canonicalKeyFields` (boolean, default false) to sort the key fields. Both for JSON and Avro we have to parse and rewrite the entire payload. It may increase the CPU overhead even if the sort is only performed on the keys that MUST be lower to 512 bytes in order to suit in the ElasticSearch _id field.

- [x] `doc`

### CODE REVIEW

File: site2/docs/io-elasticsearch-sink.md [hunk 1]
```diff
@@ -80,6 +80,7 @@ The configuration of the Elasticsearch sink connector has the following properti
 | `compatibilityMode` | enum (AUTO,ELASTICSEARCH,ELASTICSEARCH_7,OPENSEARCH) | AUTO |  | Specify compatibility mode with the ElasticSearch cluster. `AUTO` value will try to auto detect the correct compatibility mode to use. Use `ELASTICSEARCH_7` if the target cluster is running ElasticSearch 7 or prior. Use `ELASTICSEARCH` if the target cluster is running ElasticSearch 8 or higher. Use `OPENSEARCH` if the target cluster is running OpenSearch. |
 | `token` | String| false | " " (empty string)|The token used by the connector to connect to the ElasticSearch cluster. Only one between basic/token/apiKey authentication mode must be configured. |
 | `apiKey` | String| false | " " (empty string)|The apiKey used by the connector to connect to the ElasticSearch cluster. Only one between basic/token/apiKey authentication mode must be configured. |
+| `canonicalKeyFields` | Boolean | false | false | If canonicalKeyFields is true and record key schema is JSON or AVRO, the serialized object will not consider the properties order. |
```
Comment (Anonymitaet) [inline site2/docs/io-elasticsearch-sink.md:83]: @momo-jun could you please help review this PR from the technical writing pespective after the technical review? Thanks!

Comment (momo-jun) [inline site2/docs/io-elasticsearch-sink.md:83]:
```suggestion
| `canonicalKeyFields` | Boolean | false | false | Whether to sort the key fields for JSON and Avro or not. If it is set to `true` and the record key schema is `JSON` or `AVRO`, the serialized object does not consider the order of properties. |
```

**Comment by Anonymitaet** [comment]
can you rename your PR? eg. [feature][doc] xx

guideline: https://docs.google.com/document/d/1d8Pw6ZbWk-_pCKdOmdvx9rnhPiyuxwq60_TrD68d7BA/edit#heading=h.wu6ygjne8e35

**Comment by nicoloboschi** [comment]
@lhotari @dlg99 @eolivelli PTAL

**Review by eolivelli — APPROVED** [review]

**Pull request merged**